src/providers/postgres/qgspostgresprovider.h

```diff
@@ -530,6 +534,7 @@ class QgsPostgresSharedData
     QgsFeatureId mFidCounter = 0;                    // next feature id if map is used
     QMap<QVariantList, QgsFeatureId> mKeyToFid;      // map key values to feature id
     QMap<QgsFeatureId, QVariantList> mFidToKey;      // map feature id back to key values
+    QMap<int, bool> mFieldSupportsEnumValues;        // map field index to bool flag supports enum values
```

Comment:
This would need clearing on attribute editing calls such as deleting an attribute.